test/end-to-end/offer.js

```diff
@@ -1624,4 +1648,321 @@ describe("Offer endpoint tests", () => {
+                .expect(HTTPStatus.OK);
+            const res = await test_agent
+                .post(`/offers/${test_offer._id}/disable`);
+            expect(res.body).toHaveProperty("hiddenReason", OfferConstants.HiddenOfferReasons.admin);
+            expect(res.body).toHaveProperty("isHidden", true);
+        });
+
+        test("should fail if offer already disabled", async () => {
+            const res = await test_agent
+                .post(`/offers/${test_offer._id}/disable`)
+                .expect(HTTPStatus.FORBIDDEN);
+            expect(res.body).toHaveProperty("error_code", ErrorTypes.FORBIDDEN);
+            expect(res.body).toHaveProperty("errors");
+            expect(res.body.errors).toContainEqual(ValidationReasons.OFFER_HIDDEN);
+        });
+
+        test("should allow if offer hidden by default", async () => {
+            const res = await test_agent
+                .post(`/offers/${hidden_default_test_offer._id}/disable`)
+                .expect(HTTPStatus.OK);
+            expect(res.body).toHaveProperty("hiddenReason", OfferConstants.HiddenOfferReasons.admin);
+            expect(res.body).toHaveProperty("isHidden", true);
+        });
+
+        test("should allow if offer hidden by company/admin", async () => {
+            const res = await test_agent
+                .post(`/offers/${hidden_user_test_offer._id}/disable`);
+            expect(res.body).toHaveProperty("hiddenReason", OfferConstants.HiddenOfferReasons.admin);
+            expect(res.body).toHaveProperty("isHidden", true);
+        });
+    });
+
+    describe("POST /offers/:offerId/hide", () => {
+
+        let test_offer_1, test_offer_2, hidden_default_test_offer, disabled_test_offer;
+        beforeAll(async () => {
+            test_offer_1 = await Offer.create({
+                ...generateTestOffer(),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            test_offer_2 = await Offer.create({
+                ...generateTestOffer(),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+
+            hidden_default_test_offer = await Offer.create({
+                ...generateTestOffer({ isHidden: true }),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            disabled_test_offer = await Offer.create({
+                ...generateTestOffer(),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            await (new OfferService()).disable(disabled_test_offer._id, OfferConstants.HiddenOfferReasons.admin);
+        });
+
+        test("should fail if not logged in", async () => {
+            await test_agent
+                .del("/auth/login");
+
+            const res = await test_agent
+                .post(`/offers/${test_offer_1._id}/disable`)
```

Comment:
```suggestion
                .post(`/offers/${test_offer_1._id}/hide`)
```
It's hide, isn't it?